Sources/MapboxCoreNavigation/TileStore.swift

```diff
@@ -19,42 +19,43 @@ extension TileStore {
-                                                                result = result.map { $0 && value.boolValue }
-                                                            } else if expected.isError() {
-                                                                result = nil
-                                                            } else {
-                                                                assertionFailure("Unexpected value or error: \(String(describing: expected)), expected: \(NSNumber.self)")
-                                                                result = nil
-                                                            }
-                                                            
-                                                            count -= 1
-                                                            if count == 0 {
-                                                                completion(result)
-                                                            }
-                                                            lock.unlock()
-                                                         })
+                    for region in regions {
+                        self.__tileRegionContainsDescriptors(forId: region.id,
+                                                             descriptors: [latestTilesetDescriptor],
+                                                             callback: { expected in
+                            lock.lock()
+
+                            if expected.isValue(),
+                               let value = expected.value as? NSNumber {
+                                result = result.map { $0 && value.boolValue }
+                            } else if expected.isError() {
+                                result = nil
+                            } else {
+                                assertionFailure("Unexpected value or error: \(String(describing: expected)), expected: \(NSNumber.self)")
+                                result = nil
+                            }
+
+                            count -= 1
+                            if count == 0 {
+                                completion(result)
+                            }
+                            lock.unlock()
+                        })
+                    }
+                } else if expected.isError() {
+                    completion(nil)
+                } else {
+                    assertionFailure("Unexpected value or error: \(expected), expected: \(Array<TileRegion>.self)")
+                    completion(nil)
                 }
-            } else if expected.isError() {
-                completion(nil)
-            } else {
-                assertionFailure("Unexpected value or error: \(expected), expected: \(Array<TileRegion>.self)")
-                completion(nil)
             }
         }
     }
```

Comment:
`TileStore.tileRegionContainsLatestNavigationTiles` also uses `TilesetDescriptorFactory.getLatest` so I think we should refactor it the same way.